File: CSharp/OpenTraceability/Models/Events/EPCISDocument.cs
namespace OpenTraceability.Models.Events
{
    public class EPCISDocument : EPCISBaseDocument
    {
        public EPCISQueryDocument ToEPCISQueryDocument()
        {
            EPCISQueryDocument document = new EPCISQueryDocument();
            var props = typeof(EPCISBaseDocument).GetProperties();
            foreach (var p in props)
            {
                var v = p.GetValue(this);
                p.SetValue(document, v);
            }
            return document;
        }
    }
}

File: Java/OpenTraceability/src/main/opentraceability/models/events/epcisdocument.kt
package opentraceability.models.events

import opentraceability.interfaces.IEvent
import opentraceability.interfaces.IVocabularyElement
import opentraceability.models.common.StandardBusinessDocumentHeader
import java.util.*
import opentraceability.models.identifiers.*
import opentraceability.models.events.*
import java.time.OffsetDateTime

class EPCISDocument : EPCISBaseDocument() {

    fun ToEPCISQueryDocument(): EPCISQueryDocument {
        var document: EPCISQueryDocument = EPCISQueryDocument()

        //TODO: Not yet implemented
        /*
            var props = typeof(EPCISBaseDocument).GetProperties();
            foreach (var p in props)
            {
                var v = p.GetValue(this);
                p.SetValue(document, v);
            }
         */

        return document
    }

}
